hera_sim/beams.py

```diff
@@ -249,3 +249,741 @@ def interp(self, az_array, za_array, freq_array, reuse_spline=None):
+        radial_coord = za_array[np.newaxis, ...] / fscale[:, np.newaxis]
+        axial_coord = az_array[np.newaxis, ...]
+
+        # Primary beam values from Zernike polynomial
+        values = self.zernike(
+            self.beam_coeffs,
+            radial_coord * np.cos(axial_coord),
+            radial_coord * np.sin(axial_coord),
+        )
+        central_val = self.zernike(self.beam_coeffs, 0.0, 0.0)
+        values /= central_val  # ensure normalized to 1 at za=0
+
+        # Set values
+        interp_data[1, 0, 0, :, :] = values
+        interp_data[0, 0, 1, :, :] = values
+        interp_basis_vector = None
+
+        if self.beam_type == "power":
+            # Cross-multiplying feeds, adding vector components
+            pairs = [(i, j) for i in range(2) for j in range(2)]
+            power_data = np.zeros((1, 1, 4) + values.shape, dtype=float)
+            for pol_i, pair in enumerate(pairs):
+                power_data[:, :, pol_i] = (
+                    interp_data[0, :, pair[0]] * np.conj(interp_data[0, :, pair[1]])
+                ) + (interp_data[1, :, pair[0]] * np.conj(interp_data[1, :, pair[1]]))
+            interp_data = power_data
+
+        return interp_data, interp_basis_vector
+
+    def __eq__(self, other):
+        """Evaluate equality with another object."""
+        if not isinstance(other, self.__class__):
+            return False
+        return self.beam_coeffs == other.beam_coeffs
+
+    def zernike(self, coeffs, x, y):
+        """
+        Zernike polynomials (up to degree 66) on the unit disc.
+
+        This code was adapted from:
+        https://gitlab.nrao.edu/pjaganna/zcpb/-/blob/master/zernikeAperture.py
+
+        Parameters
+        ----------
+        coeffs : array_like
+            Array of real coefficients of the Zernike polynomials, from 0..66.
+
+        x, y : array_like
+            Points on the unit disc.
+
+        Returns
+        -------
+        zernike : array_like
+            Values of the Zernike polynomial at the input x,y points.
+        """
+        # Coefficients
+        assert len(coeffs) <= 66, "Max. number of coeffs is 66."
+        c = np.zeros(66)
+        c[: len(coeffs)] += coeffs
+
+        # x2 = self.powl(x, 2)
+        # y2 = self.powl(y, 2)
+        x2, x3, x4, x5, x6, x7, x8, x9, x10 = (
+            x ** 2.0,
+            x ** 3.0,
+            x ** 4.0,
+            x ** 5.0,
+            x ** 6.0,
+            x ** 7.0,
+            x ** 8.0,
+            x ** 9.0,
+            x ** 10.0,
+        )
+        y2, y3, y4, y5, y6, y7, y8, y9, y10 = (
+            y ** 2.0,
+            y ** 3.0,
+            y ** 4.0,
+            y ** 5.0,
+            y ** 6.0,
+            y ** 7.0,
+            y ** 8.0,
+            y ** 9.0,
+            y ** 10.0,
+        )
+
+        # Setting the equations for the Zernike polynomials
+        # r = np.sqrt(powl(x,2) + powl(y,2))
+        Z1 = c[0] * 1  # m = 0    n = 0
```

Comment:
formatting-wise, I think it might be easier to set z as a dict, like

```python
z = {
    1: c[0] * 1,
    2: ...
}
```

Then the final thing can just be `sum(z.values())`. The above formatting makes it easy to identify a particular coefficient still.